src/sentry/search/events/builder/metrics.py

```diff
@@ -147,11 +158,18 @@ def _on_demand_metric_spec_map(self) -> Optional[Dict[str, OnDemandMetricSpec]]:
         return {
```

Comment:
I think this whole return can be reduced to:
```return {col: self._get_on_demand_metric_spec(col) for col in self._get_aggregates()}```

Perhaps we need to check `_get_on_demand_metric_spec(col)` within `_get_aggregates`.

Reply:
yeah, maybe we can rename `_get_aggregates` to `_get_ondemand_fields` then it would make more sense to me imo, we should avoid any ondemand specific code if we're calling it `aggregates`

Reply:
Like this, right?

```
def _get_ondemand_fields(self) -> list[str]:
        return [c for c in self.selected_columns if fields.is_function(c) and self._get_on_demand_metric_spec(c)]
```